mat64/io_test.go

```diff
@@ -276,3 +276,67 @@ func TestVectorIORoundTrip(t *testing.T) {
 		}
 	}
 }
+
+func BenchmarkMarshalDense10(b *testing.B)    { marshalBinaryBenchDense(b, 10) }
+func BenchmarkMarshalDense100(b *testing.B)   { marshalBinaryBenchDense(b, 100) }
+func BenchmarkMarshalDense1000(b *testing.B)  { marshalBinaryBenchDense(b, 1000) }
+func BenchmarkMarshalDense10000(b *testing.B) { marshalBinaryBenchDense(b, 10000) }
+
+func marshalBinaryBenchDense(b *testing.B, size int) {
+	m := NewDense(1, size, make([]float64, size))
```

Comment:
I replaced the slice of 0 with a linrange:

```
name                    old time/op    new time/op    delta
MarshalDense10-4          2.64µs ± 1%    2.64µs ± 1%   ~             (p=0.635 n=5+5)
MarshalDense100-4         22.1µs ± 1%    22.1µs ± 1%   ~             (p=0.548 n=5+5)
MarshalDense1000-4         216µs ± 1%     216µs ± 1%   ~             (p=0.841 n=5+5)
MarshalDense10000-4       2.15ms ± 0%    2.15ms ± 1%   ~             (p=0.151 n=5+5)
UnmarshalDense10-4        1.84µs ± 1%    1.83µs ± 1%   ~             (p=0.278 n=5+5)
UnmarshalDense100-4       15.4µs ± 1%    15.4µs ± 1%   ~             (p=0.841 n=5+5)
UnmarshalDense1000-4       152µs ± 1%     152µs ± 1%   ~             (p=0.690 n=5+5)
UnmarshalDense10000-4     1.51ms ± 1%    1.51ms ± 1%   ~             (p=0.841 n=5+5)
MarshalVector10-4         2.52µs ± 0%    2.51µs ± 1%   ~             (p=0.079 n=5+5)
MarshalVector100-4        22.3µs ± 0%    22.3µs ± 1%   ~             (p=0.690 n=5+5)
MarshalVector1000-4        220µs ± 1%     219µs ± 1%   ~             (p=0.421 n=5+5)
MarshalVector10000-4      2.20ms ± 1%    2.19ms ± 0%   ~             (p=0.421 n=5+5)
UnmarshalVector10-4       1.75µs ± 1%    1.74µs ± 1%   ~             (p=0.389 n=5+5)
UnmarshalVector100-4      15.5µs ± 1%    15.5µs ± 1%   ~             (p=1.000 n=5+5)
UnmarshalVector1000-4      152µs ± 0%     152µs ± 1%   ~             (p=0.690 n=5+5)
UnmarshalVector10000-4    1.52ms ± 1%    1.52ms ± 1%   ~             (p=0.841 n=5+5)

name                    old alloc/op   new alloc/op   delta
MarshalDense10-4            400B ± 0%      400B ± 0%   ~     (all samples are equal)
MarshalDense100-4         2.64kB ± 0%    2.64kB ± 0%   ~     (all samples are equal)
MarshalDense1000-4        24.3kB ± 0%    24.3kB ± 0%   ~     (all samples are equal)
MarshalDense10000-4        242kB ± 0%     242kB ± 0%   ~             (p=1.000 n=5+5)
UnmarshalDense10-4          240B ± 0%      240B ± 0%   ~     (all samples are equal)
UnmarshalDense100-4       1.78kB ± 0%    1.78kB ± 0%   ~     (all samples are equal)
UnmarshalDense1000-4      16.3kB ± 0%    16.3kB ± 0%   ~     (all samples are equal)
UnmarshalDense10000-4      162kB ± 0%     162kB ± 0%   ~     (all samples are equal)
MarshalVector10-4           384B ± 0%      384B ± 0%   ~     (all samples are equal)
MarshalVector100-4        2.62kB ± 0%    2.62kB ± 0%   ~     (all samples are equal)
MarshalVector1000-4       24.3kB ± 0%    24.3kB ± 0%   ~     (all samples are equal)
MarshalVector10000-4       242kB ± 0%     242kB ± 0%   ~     (all samples are equal)
UnmarshalVector10-4         224B ± 0%      224B ± 0%   ~     (all samples are equal)
UnmarshalVector100-4      1.76kB ± 0%    1.76kB ± 0%   ~     (all samples are equal)
UnmarshalVector1000-4     16.3kB ± 0%    16.3kB ± 0%   ~     (all samples are equal)
UnmarshalVector10000-4     162kB ± 0%     162kB ± 0%   ~     (all samples are equal)

name                    old allocs/op  new allocs/op  delta
MarshalDense10-4            26.0 ± 0%      26.0 ± 0%   ~     (all samples are equal)
MarshalDense100-4            206 ± 0%       206 ± 0%   ~     (all samples are equal)
MarshalDense1000-4         2.01k ± 0%     2.01k ± 0%   ~     (all samples are equal)
MarshalDense10000-4        20.0k ± 0%     20.0k ± 0%   ~     (all samples are equal)
UnmarshalDense10-4          16.0 ± 0%      16.0 ± 0%   ~     (all samples are equal)
UnmarshalDense100-4          106 ± 0%       106 ± 0%   ~     (all samples are equal)
UnmarshalDense1000-4       1.01k ± 0%     1.01k ± 0%   ~     (all samples are equal)
UnmarshalDense10000-4      10.0k ± 0%     10.0k ± 0%   ~     (all samples are equal)
MarshalVector10-4           24.0 ± 0%      24.0 ± 0%   ~     (all samples are equal)
MarshalVector100-4           204 ± 0%       204 ± 0%   ~     (all samples are equal)
MarshalVector1000-4        2.00k ± 0%     2.00k ± 0%   ~     (all samples are equal)
MarshalVector10000-4       20.0k ± 0%     20.0k ± 0%   ~     (all samples are equal)
UnmarshalVector10-4         14.0 ± 0%      14.0 ± 0%   ~     (all samples are equal)
UnmarshalVector100-4         104 ± 0%       104 ± 0%   ~     (all samples are equal)
UnmarshalVector1000-4      1.00k ± 0%     1.00k ± 0%   ~     (all samples are equal)
UnmarshalVector10000-4     10.0k ± 0%     10.0k ± 0%   ~     (all samples are equal)
```

it's probably less confusing to have actual data rather than zeros anyway (so I left the linrange)